src/__tests__/transpiled-code.ts

```diff
@@ -0,0 +1,152 @@
+import { stripIndent } from 'common-tags'
+import { mockContext } from '../mocks/context'
+import { parse } from '../parser'
+import { transpile } from '../transpiler'
+
+// DO NOT HAVE 'native' AS A SUBSTRING IN CODE STRINGS ANYWHERE IN THIS FILE!
+function expectTranspiledToMatchSnapshot(code: string) {
+  return () => {
+    const context = mockContext(4)
+    const transpiled = transpile(parse(code, context)!, context.contextId).transpiled
+    // replace native[<number>] as they may be inconsistent
+    const replacedNative = transpiled.replace(/native\[\d+]/g, 'native')
+    // replace the line hiding globals as they may differ between environments
+    const replacedGlobalsLine = replacedNative.replace(/\n\(\(.*\)/, '\n(( <globals redacted> )')
+    // replace declaration of builtins since they're repetitive
+    const replacedBuiltins = replacedGlobalsLine.replace(
+      /\n *const \w+ = native\.builtins\.get\("\w+"\);/g,
+      ''
+    )
+    expect({ code, transpiled: replacedBuiltins }).toMatchSnapshot()
```

Comment:
Maybe better to return the promise and let the client code do the toMatchSnapshot(). Same amount of typing in both cases

Reply:
What do you mean by this?

`return expect(..)` and then do `expectTranspiled('code').toMatchSnapshot()`?

Reply:
Yep, exactly

Reply:
Looking back, I seem to have standardised `test*` which returns a promise that returns objects (which may or may not secretly check things with expect, e.g. status is successful), `snapshot*` which returns a similar promise but snapshots things first, and `expect*` which returns an expect object instantiated with one of the fields (and also may or may not secretly check other things with expect)